kcbq-connector/src/main/java/com/wepay/kafka/connect/bigquery/write/batch/TableWriter.java

```diff
@@ -85,6 +85,8 @@ public void run() {
           if (isBatchSizeError(err)) {
             failureCount++;
             currentBatchSize = getNewBatchSize(currentBatchSize);
+          } else {
+            throw err;
```

Comment:
Why is this required?

Reply:
If we can also have a check for it not being a temporary network error (`backendError`) before throwing it, it might help

https://cloud.google.com/bigquery/docs/error-messages

Reply:
It's already checked in `BigQueryWriter`.